tools/tcprtt.py

```diff
@@ -79,18 +83,34 @@
     u64 slot;
 } sock_key_t;
 
-STORAGE
+typedef struct sock_latenty {
+    u64 latency;
+    u64 count;
+} sock_latency_t;
+
+BPF_HISTOGRAM(hist_srtt, sock_key_t);
+BPF_HASH(latency, u64, sock_latency_t);
 
 int trace_tcp_rcv(struct pt_regs *ctx, struct sock *sk, struct sk_buff *skb)
 {
     struct tcp_sock *ts = tcp_sk(sk);
     u32 srtt = ts->srtt_us >> 3;
     const struct inet_sock *inet = inet_sk(sk);
+
+    /* filters */
     u16 sport = 0;
     u16 dport = 0;
     u32 saddr = 0;
     u32 daddr = 0;
 
+    /* for histogram */
+    sock_key_t key;
+
+    /* for avg latency, if no saddr/daddr specified, use 0(addr) as key */
+    u64 addr = 0;
+    sock_latency_t newlat = {0};
+    sock_latency_t *lat;
```

Comment:
I think you can put `lat` and `newlat` into STORE_LATENCY macro. to make things easier.